PLT-1459: Updated README of marconi-chain-index and marconi-sidechain

koslambrou opened the pull request:
* Rename the old `marconi-chain-index` README.md to README_MARCONI_DRAFT as it represents the README for the Marconi project as a whole. Will probably be used elsewhere.

* Update the `marconi-sidechain` README with latest changes and added some instructions on how to build it.

* Create a README for `marconi-chain-index` with a similar format to `marconi-sidechain` as they share many similar things. This will be worked on more in the future.

Note: The diff in the `marconi-chain-index/README.md` can be glanced over very quickly, as it's basically the same as the `marconi-sidechain/README.md` (minus the JSON-RPC server)

<!--
Here are some checklists you may like to use. Use your judgement.

This is just a checklist, all the normative suggestions are covered in more detail in CONTRIBUTING.
-->
Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [ ] Commit sequence broadly makes sense
    - [ ] Key commits have useful messages
    - [ ] Formatting, PNG optimization, etc. are updated
    - [ ] Important changes are reflected in changelog.d of the affected packages
- PR
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description
    - [ ] Reference the ADR in the PR and reference the PR in the ADR (if revelant)
    - [ ] Reviewer requested

kayvank commented on marconi-sidechain/README.md:
```diff
@@ -29,173 +28,199 @@ The interface for marconi-sidechain uses [JSON-RPC](http://www.simple-is-better.
+* you have a local running cardano-node instance.
+* you've set the following environment variables:
+  * `CARDANO_NODE_SOCKET_PATH`: socket path of your local cardano-node instance
+  * `MARCONI_DB_DIRECTORY`: directoy in which to create the various SQLite database files
```
direcory

eyeinsky commented on marconi-chain-index/README.md:
````diff
@@ -1,178 +1,99 @@
-   * Konstantinos has some for Plutus Apps.
-   * Radu has some for Marconi.
+```sh
+marconi-chain-index \
````
I'm not a fan of the golden tests that we have for the cli, but now that we do have them let's link to them?

koslambrou replied:
We don't have golden tests for the `marconi-chain-index` CLI, and it's a bit different than the `marconi-sidechain` one.

koslambrou replied:
I made an argument here https://github.com/input-output-hk/plutus-apps/pull/960#discussion_r1109924411 about why we would want golden tests.

What don't you like exactly? What can we do different which can achieve the same benefits as the golden tests that we have.

kayvank replied:
> We don't have golden tests for the `marconi-chain-index` CLI, and it's a bit different than the `marconi-sidechain` one.

We added CLI golden test to [marconi-shain-index ](https://github.com/input-output-hk/plutus-apps/blob/main/marconi-chain-index/test/Spec/Golden/Cli/marconi-chain-index___help.help)

koslambrou replied:
@kayvank Thanks, added a link to the golden file. However, in this case, I want to show an example off how to start the command in the README, and the golden file doesn't provide that. We can probably extend it, but for now, let's keep it as is.

eyeinsky replied:
Regarding not liking golden tests for CLI: the CLI text that an executable prints are directly generated from the CLI definitions that we defined by the help of optparse-applicative library and tests for that process already exist in optparse-applicative. I.e if we don't want to change the CLI then we should just ...not change the CLI definitions. And when we *do* want to change the CLI definitions then we now have *two* files to change.

eyeinsky commented on marconi-sidechain/README.md:
```diff
@@ -29,173 +28,199 @@ The interface for marconi-sidechain uses [JSON-RPC](http://www.simple-is-better.
+By default, all addresses are indexed in the database.
 
-Compatible version of [cardano-node](https://github.com/input-output-hk/plutus-apps/blob/main/cabal.project#L246) must be running with the socket-path as outlined below:
+Some example addresses you can for pre-production-testnet are:
```
you can => you can use

eyeinsky commented:
I found the readme that is now renamed to draft useful when I first joined as it gave an overview of what marconi did. While the new parts on how to build and run is also useful, then maybe they could be merged instead -- as the new readme is brief on what marconi does, what it's architecture is, etc.

koslambrou commented on marconi-chain-index/README.md:
```diff
@@ -1,178 +1,99 @@
 
-## How to Install and Configure
+* from the last chainpoint (if none, from genesis), fetch blocks from the local node, extract required data and index them in the database.
+* run a JSON-RPC server which will listen for any queries on the indexed data.
```
Copy paste error. Thanks :)

koslambrou commented:
> I found the readme that is now renamed to draft useful when I first joined as it gave an overview of what marconi did. While the new parts on how to build and run is also useful, then maybe they could be merged instead -- as the new readme is brief on what marconi does, what it's architecture is, etc.

@eyeinsky I agree that it's very valuable. However, it is **not** the README for `marconi-chain-index`, but README for the Marconi project as a whole. That draft README would essentially be the README for the marconi project once we split it in a separate repo.
